openmdao/core/total_jac.py

```diff
@@ -1387,7 +1388,13 @@ def compute_totals(self):
                                                                                   mode=self.mode):
                                     print("   {}".format(local_ind))
                             else:
-                                print("('{0}', [{1}])".format(key, inds))
+                                print_key = key
+                                if key.startswith('_auto_ivc'):
+                                    conns = model._problem_meta['connections']
+                                    for src, tgt in conns.items():
```

Comment:
I know this is only happening if debug_print is True, but it's a really slow way to determine the target. Might want to put an `if debug_print` block at the top outside of all the loops and create a reverse mapping dict there.  Also, should we consider, once we have an absolute input name, mapping that to the promoted name since that's what we use in other user-facing places where we're dealing with auto_ivc connected vars.

Reply:
Added a cache for the name translations.